### ISSUE TITLE
[Ruby] NoMethodError undefined method `length' for -1:Integer

### ISSUE THREAD

**Issue by chrismytton** [paulfitz/daff#97]
I'm getting the following error when using the Ruby version of daff (1.3.19), installed with `gem install daff`.

```
$ daff.rb query.csv morph.csv                  
/Users/chris/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/daff-1.3.19/lib/lib/coopy/table_diff.rb:399:in `setup_moves': undefined method `length' for -1:Integer (NoMethodError)                                                            
        from /Users/chris/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/daff-1.3.19/lib/lib/coopy/table_diff.rb:976:in `hilite_single'                                                                                                       
        from /Users/chris/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/daff-1.3.19/lib/lib/coopy/table_diff.rb:1079:in `hilite_with_nesting'                                                                                                
        from /Users/chris/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/daff-1.3.19/lib/lib/coopy/coopy.rb:279:in `run_diff'                                                                                                                 
        from /Users/chris/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/daff-1.3.19/lib/lib/coopy/coopy.rb:803:in `run'                                                                                                                      
        from /Users/chris/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/daff-1.3.19/lib/lib/coopy/coopy.rb:831:in `coopyhx'                                                                                                                  
        from /Users/chris/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/daff-1.3.19/lib/lib/coopy/coopy.rb:948:in `main'                                                                                                                     
        from /Users/chris/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/daff-1.3.19/bin/daff.rb:3:in `<top (required)>'                                                                                                                      
        from /Users/chris/.rbenv/versions/2.4.1/bin/daff.rb:22:in `load'                                               
        from /Users/chris/.rbenv/versions/2.4.1/bin/daff.rb:22:in `<main>'                                             
```

The two CSV files I used above, `query.csv` and `morph.csv`, are available [in this gist](https://gist.github.com/chrismytton/683058d7a476ca9c55f2399c093885b5).

I've tried running those CSV files through the node version, and that works as expected. [Output from node command is here](https://gist.github.com/chrismytton/9e12fcbb550cc0c5395b5b798e29475c).

Also worth noting that I've seen a few variations on this error, it's not always `-1:Integer`, I've seen other numbers there, both positive and negative, when testing re-ordering the CSV to try and work around the error.

I'm very happy to help with some coding/debugging, but I'm not very familiar with Haxe. So any suggestions or pointers are welcome!

---

p.s. Many thanks for releasing this great open source project, @paulfitz! We're using it as part of [EveryPolitician](http://everypolitician.org/) and it's been super useful so far! ❤️

**Comment by paulfitz** [comment]
Thanks for reporting and giving .csv files @chrismytton.  I've replicated the problem.  Will check what is up.  EveryPolitician looks nifty btw!

**Comment by mhl** [comment]
I tried reproducing this, and it works fine with current master (2106718d78674b) but fails at the version tagged v1.3.19 (0b6742b73131c7783d681e6855fc8f86b2c09549), which I think is what the gem was built from.

Using git bisect, it seems that the commit that fixed the problem is 3967f10f0f2829616d5 - "work around weakness in ruby translation".

@paulfitz I think from the above that if you were to create a new version, 1.3.26, based on current master and released a new version of the gem based on that, it would fix the problem for us - might that be possible?

(P.S. @chrismytton and I work on the same team.)

**Comment by paulfitz** [comment]
@mhl you're totally right, this turned out to be a solved problem, I just hadn't pushed an updated gem.  Can you check if the now-published 1.3.26 works for you?

**Comment by chrismytton** [comment]
@paulfitz @mhl Yep, that's done the trick. Thanks both for taking the time to look into this! 💖
